keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -413,28 +412,83 @@ On Pod Create/Update/Delete:
 
 ### Kube-Proxy
 
-Kube-proxy will be modified to consume EndpointSlice instances besides Endpoints resource. A flag will be added to kube-proxy to toggle the mode. 
-
-```
-Watch: Service, EndpointSlice ==> Manage: iptables, ipvs, etc
-```
-- Merge multiple EndpointSlice into an aggregated list.
-- Reuse the existing processing logic 
-
-### Endpoint Controller (classic)
-
-In order to ensure backward compatibility for external consumer of the core/v1 Endpoints API, the existing K8s endpoint controller will keep running until the API is EOL. The following limitations will apply:
-- Starting from EndpointSlice beta: If # of endpoints in one Endpoints object exceed 1000, generate a warning event to the object. 
-- Starting from EndpointSlice GA: Only include up to 1000 endpoints in one Endpoints Object and throw events.
+Kube-Proxy will support consuming EndpointSlice resources as an alternative to
+Endpoints resources. This will be enabled with the `EndpointSliceProxying`
+feature gate.
+
+This will require minimal changes to proxier implementations, instead updating
+the shared code that transforms Endpoints into a generic data structure to also
+transform EndpointSlices into that same data structure.
+
+### Endpoints Controller (classic)
+
+In order to ensure backwards compatibility for external consumer of the core/v1
+Endpoints API, the existing Endpoints controller will keep running until the
+API is EOL. As EndpointSlices become more stable, the Endpoints controller will
+gradually limit functionality.
+
+* Kubernetes 1.19: If the number of endpoints in one Endpoints object exceeds
+  1000, generated a warning event on the Endpoints resource.
+* Kubernetes 1.20: Limit the number of endpoints in one Endpoints object to
+  1000 and generate warning events on Endpoints resource if additional endpoints
+  are dropped.
+
+### EndpointsMirroring Controller
+
+In some cases, custom Endpoints resources are created by applications. To ensure
+that these applications will not need to immediately update their
+implementations to also create custome EndpointSlices, a new EndpointsMirroring
+controller will be used to mirror custome Endpoints resources to corresponding
+EndpointSlices.
+
+* A new `endpointsmirroring.kubernetes.io/skip` label will be introduced. When
```

Comment:
a label that includes "endpointslice" might be more helpful in understanding what the label connects to

Reply:
Updated to `endpointslicemirroring` everywhere. I don't really love that name, and it is really long, but it is likely more clear. I originally chose `endpointsmirroring` because it was the Endpoints resources that were being mirrored, but that's not particularly clear. I agree that having `endpointslice` in the names here will make the purpose of the controller/label easier to understand.